CBLClient/Apps/CBLTestServer-Java-Desktop/src/main/java/com/couchbase/mobiletestkit/javatestserver/TestServerContext.java

```diff
@@ -0,0 +1,104 @@
+package com.couchbase.mobiletestkit.javatestserver;
+
+import com.couchbase.mobiletestkit.javacommon.Context;
+import com.couchbase.mobiletestkit.javacommon.util.Log;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.*;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Base64;
+
+public class TestServerContext implements Context {
+    private final String TAG = "JavaContext";
+    private File directory;
+
+    public TestServerContext(File directory) {
+        this.directory = directory;
+    }
+
+    @Override
+    public File getFilesDir() {
+        return this.directory;
+    }
+
+
+    @Override
+    public InputStream getAsset(String name) {
+        return TestServerMain.class.getResourceAsStream("/" + name);
+    }
+
+    @Override
+    public String getPlatform() {
+        return "java";
+    }
+
+    @Override
+    public String getLocalIpAddress() {
+        String ip = "";
```

Comment:
Why ""?  Why not `null`?

Reply:
implementation for this method has completed changed, variable "ip" is no longer defined